File: db4on/core/src/core/com/db4o/TreeString.cs
namespace com.db4o
{
	/// <exclude></exclude>
	public class TreeString : com.db4o.foundation.Tree
	{
		public string _key;

		public TreeString(string a_key)
		{
			this._key = a_key;
		}

		protected override com.db4o.foundation.Tree ShallowCloneInternal(com.db4o.foundation.Tree
			 tree)
		{
			com.db4o.TreeString ts = (com.db4o.TreeString)base.ShallowCloneInternal(tree);
			ts._key = _key;
			return ts;
		}

		public override object ShallowClone()
		{
			return ShallowCloneInternal(new com.db4o.TreeString(_key));
		}

		public override int Compare(com.db4o.foundation.Tree a_to)
		{
			return com.db4o.YapString.Compare(com.db4o.YapConst.stringIO.Write(((com.db4o.TreeString
				)a_to)._key), com.db4o.YapConst.stringIO.Write(_key));
		}
	}
}

File: db4oj/core/src/com/db4o/TreeString.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;


/**
 * @exclude
 */
public class TreeString extends Tree{
    
	public final String i_key;
	
	public TreeString(String a_key){
		this.i_key = a_key;
	}

    int compare(Tree a_to) {
        return YapString.compare(YapConst.stringIO.write(((TreeString)a_to).i_key), YapConst.stringIO.write(i_key));
    }

}
